From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Thu, 5 Jan 2012 00:57:49 +0100
Subject: widget-export-actions-location

Install libgammaray_widget_export_actions.so to /usr/lib/gammaray
as it is application private library. Fixes lintian warnings/errors:

package-name-doesnt-match-sonames
shlib-without-versioned-soname
postinst-must-call-ldconfig
---
 CMakeLists.txt                                 |    1 +
 core/tools/widgetinspector/CMakeLists.txt      |    2 +-
 core/tools/widgetinspector/widgetinspector.cpp |    2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74548df..971ecbf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,6 +87,7 @@ endif()
 set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
 set(LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
 set(PLUGIN_INSTALL_DIR "${LIB_INSTALL_DIR}/qt4/plugins")
+set(MODULE_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray")
 
 set(
   INSTALL_TARGETS_DEFAULT_ARGS
diff --git a/core/tools/widgetinspector/CMakeLists.txt b/core/tools/widgetinspector/CMakeLists.txt
index eefa624..e851f8d 100644
--- a/core/tools/widgetinspector/CMakeLists.txt
+++ b/core/tools/widgetinspector/CMakeLists.txt
@@ -10,4 +10,4 @@ target_link_libraries(gammaray_widget_export_actions
   ${QT_QTDESIGNER_LIBRARIES}
 )
 
-install(TARGETS gammaray_widget_export_actions ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS gammaray_widget_export_actions DESTINATION ${MODULE_INSTALL_DIR})
diff --git a/core/tools/widgetinspector/widgetinspector.cpp b/core/tools/widgetinspector/widgetinspector.cpp
index dcd9e1a..1fafc92 100644
--- a/core/tools/widgetinspector/widgetinspector.cpp
+++ b/core/tools/widgetinspector/widgetinspector.cpp
@@ -246,7 +246,7 @@ void WidgetInspector::callExternalExportAction(const char *name,
 {
   if (!m_externalExportActions.isLoaded()) {
     const QString probePath =
-      QString::fromLocal8Bit(qgetenv("GAMMARAY_PROBE_PATH"));
+      QString::fromLocal8Bit(qgetenv("GAMMARAY_PROBE_PATH") + "/gammaray");
 
     m_externalExportActions.setFileName(
       probePath + QLatin1String("/libgammaray_widget_export_actions"));
